### ISSUE TITLE
Checkbox selector: "select all" not selecting all records in grid

### ISSUE THREAD

**Issue by amallikage** [SitePen/dgrid#763]
Hello;

I'm experiencing an issue with the 'select all' checkbox in a grid. The 'select all' button does not appear to select all records in the grid. The issue can be observed on this sample page: http://dojofoundation.org/packages/dgrid/js/dgrid/test/selector.html

Enable the browser console (F12 in Chrome) and select the 'select all' button on the first grid on the page (under the heading: "A grid with checkbox selectors, with select-all enabled (top checkbox, and Ctrl/Cmd+A)"). In the console output you can see that only the values up to ID 23 are being selected:

From console "grid: selected:  0, 1, 2, 4, 5, 7, 8, 9, 11, 12, 14, 15, 16, 18, 19, 21, 22, 23"

If you refresh the page, scroll through to the bottom of the grid and then select 'select all' you'll see that all records are selected:

From console: grid: selected (via click):  0, 1, 2, 4, 5, 7, 8, 9, 11, 12, 14, 15, 16, 18, 19, 21, 22, 23, 25, 26, 28, 29, 30, 32, 33, 35, 36, 37, 39, 40, 42, 43, 44, 46, 47, 49, 50, 51, 53, 54, 56, 57, 58, 60, 61, 63, 64, 65, 67, 68, 70, 71, 72, 74, 75, 77, 78, 79, 81, 82, 84, 85, 86, 88, 89, 91, 92, 93, 95, 96, 98, 99

Anyone have an ideas on how this can be corrected, or if there's a workaround? I need to be able to capture all ID's in a grid without having to scroll through the results first.

Any help would be greatly appreciated.

Thanks.

**Comment by chr1st0scli** [comment]
Hi. I have a reproduction of what allenerb reports. The call to selectAll skips some rows in the middle. I am attaching a screenshot and I am here for further details. The skipping rows pattern repeats again further down the grid (notice the scrollbar). Note that I have 137 rows loaded in my grid. I am not using the latest version, I can't find which one I am using. The only thing I found in a README.md file is that I have the dgrid version which is recommended for use with Dojo 1.8.1. Thank you.
![skippingrows](https://cloud.githubusercontent.com/assets/5046010/8327468/59cbfa90-1a71-11e5-88e7-ec66fa7f24a9.png)

**Comment by tushkyparikh** [comment]
Hi 
I am facing the same issue, Can you please let me know if there's any workaround for this?
Any help would be greatly appreciated.

Thanks.

**Comment by chr1st0scli** [comment]
Hi, this is what I did for amallikage's case as described by him. For cases in which I needed a grid with check boxes and a "select all" functionality, I simply used grids with no scroll bars, which means that all data was loaded at once (no lazy-loading via scrolling down). In my case, this was acceptable as my grids were not very loaded. For more data, you might want to consider the aforementioned solutions.
